### PR TITLE
Parameter manager merge branch

### PR DESCRIPTION
**Pull request by javmorcas**
This pull request is coming from the issue #XX. When the pull request is merged into "master" the issue will be [resolved, closed, fixed] #XX

### CODE REVIEW

**Comment by javmorcas** [comment]
@alfonso-soto, @softwarejimenez, unless you think it is needed, I was planning on considering this feature as internal, therefore I am not going to create specific documentation for this.

File: srcCpp/RTIDDSImpl.cxx [hunk 1]
```diff
@@ -2723,15 +2270,17 @@ unsigned long RTIDDSImpl<T>::GetInitializationSampleCount()
...
      * If we are using batching we need to take into account tha the Send Queue
      * will be per-batch, therefore for the number of samples:
      */
-    if (_BatchSize > 0) {
+    if (_PM->get<long>("batchSize") > 0) {
         initializeSampleCount = (std::max)(
-                _SendQueueSize * (_BatchSize / _DataLen),
+                _PM->get<int>("sendQueueSize") *
+                        (_PM->get<long>("batchSize") /
+                        _PM->get<unsigned long>("dataLen")),
```
Comment (alfonso-soto) [inline srcCpp/RTIDDSImpl.cxx]: This need to be changed as well as RawTransport Branch.

But if we change it to 
`_PM->get<unsigned long long>("dataLen")),`

Compiling with gcc 7.3.0 I got the following error:
```
RTIDDSImpl.cxx: In member function ‘long unsigned int RTIDDSImpl<T>::GetInitializationSampleCount()’:
RTIDDSImpl.cxx:2284:38: error: no matching function for call to ‘max(long long unsigned int, long unsigned int&)’
                 initializeSampleCount);
```
We need to add a new casting for **initializeSampleCount**  
So, in the end, this needs to be like:
```
        initializeSampleCount = (std::max)(
                _PM->get<int>("sendQueueSize") *
                        (_PM->get<long>("batchSize") /
                        _PM->get<unsigned long long>("dataLen")),
                (unsigned long long) initializeSampleCount);
```

Comment (javmorcas) [reply srcCpp/RTIDDSImpl.cxx]: Fixed

File: srcCpp/perftest_publisher.cxx [hunk 2]
```diff
@@ -1921,7 +1437,7 @@ int perftest_cpp::Publisher()
                 fprintf(stderr, "Problem creating ReceiverThread for LatencyReadThread.\n");
                 return -1;
             }
-            
+
```
Comment (softwarejimenez) [inline srcCpp/perftest_publisher.cxx]: remove extra line

File: srcCppCommon/Parameter.h [hunk 3]
```diff
@@ -0,0 +1,329 @@
...
+        {
+            return static_cast<ParameterPair<K, V>*>(_param);
+        }
+
+        AnyParameter& operator=(AnyParameter &other)
+        {
+            _param = other._param;
+            other._param = NULL;
+            return *this;
+        };
```
Comment (softwarejimenez) [inline srcCppCommon/Parameter.h]: remove  ;

File: .gitignore [hunk 4]
```diff
@@ -27,3 +27,11 @@
...
 doc
 srcDoc/_build
+
+#Files copied by the build Script.
+/srcCpp/Parameter*
+/srcCpp03/Parameter*
+/srcCpp03/CpuMonitor*
+/srcCpp/CpuMonitor*
+/srcCpp/perftestThreadPriorities.*
+/srcCpp03/perftestThreadPriorities.*
```
[the remark this answers is not in the document]
Comment (javmorcas) [reply .gitignore]: Done

File: srcCpp/RTIDDSImpl.cxx [hunk 5]
```diff
@@ -741,63 +189,106 @@ bool RTIDDSImpl<T>::ParseConfig(int argc, char *argv[])
...
-    if (_isPublisher && _useCft) {
-        fprintf(stderr,
-                "Content Filtered Topic is not a parameter in the publisher side.\n");
-    }
 
-    if(!_transport.parseTransportOptions(argc, argv)) {
-        fprintf(stderr,
-                "Failure parsing the transport options.\n");
+    // Manage transport parameter
+    if(!_transport.validate_input()) {
```
[the remark this answers is not in the document]
Comment (javmorcas) [reply srcCpp/RTIDDSImpl.cxx]: Done. Also for Cpp03

File: srcCpp/RTIDDSImpl.cxx [hunk 6]
```diff
@@ -142,596 +132,54 @@ void RTIDDSImpl<T>::Shutdown()
...
-            "\t-securePrivateKey <file>      - Private key file <optional>\n" +
-            "\t                                Default: \"./resource/secure/subkey.pem\"\n";
-  #endif
-
-    fprintf(stderr, "%s", usage_string.c_str());
-}
 
-/*********************************************************
- * ParseConfig
+ * Validate and manage the parameterS
```
[the remark this answers is not in the document]
Comment (javmorcas) [reply srcCpp/RTIDDSImpl.cxx]: Done

File: srcCppCommon/Parameter.h (hunk 3 above)
Comment (javmorcas) [reply srcCppCommon/Parameter.h]: Done

**Pull request merged**